packages/astro/src/content/types-generator.ts

```diff
@@ -155,7 +157,16 @@ export async function createContentTypesGenerator({
 			return { shouldGenerateTypes: false };
 		}
 
-		const { id, slug, collection } = entryInfo;
+		// `slug` may be present in entry frontmatter.
+		// This should be respected by the generated `slug` type!
+		// Parse frontmatter and retrieve `slug` value for this.
+		// Note: will raise any YAML exceptions and `slug` parse errors (i.e. `slug` is a boolean)
+		// on dev server startup or production build init.
+		const rawContents = await fs.promises.readFile(event.entry, 'utf-8');
```

Comment:
Is this going to run when the file is first created? If so are we taking into account that the file is empty and there is no frontmatter?

Reply:
Ah, yep it will! We should have an empty case handler, good catch. Will make sure an error isn't thrown 👍

Reply:
Looks like the empty case is handled fine. _Did_ help catch an error when deleting a file though. Thanks for that!